### DISCUSSION TITLE
GoPro Hero 10 USB Trigger

### DISCUSSION THREAD

**Discussion by giant-killer-racing** [gopro/labs#1006] [General]
Hi All, 
I am trying to use a USB trigger to do the following: Labs firmware installed:

Wake the camera when USB power is detected (and battery is installed)
Record to default setting automatically
Save file from battery when USB voltage is removed

I have used a few scripts and tested functions independent of each other. The issue I am having is that my camera wont auto wake on USB trigger. I am trying to use this one from Aerial Trigger on YouTube as it appears to be what I am looking for. 

https://www.youtube.com/watch?v=FajSd4N-jrc&lc=UgyqbrNu2S-lrG2Ok0N4AaABAg.A6V3I-ZfxHkA6WgHjnlH6T

!uN!S!u10E!R
<img width="745" alt="Screenshot 2024-07-30 at 11 28 55 AM" src="https://github.com/user-attachments/assets/628eef82-43d8-4f97-bfd7-d651ea40ba70">

Any setting on the 10 that would be preventing this? 

This is my attempt at block code: 

<img width="263" alt="Screenshot 2024-07-30 at 12 06 47 PM" src="https://github.com/user-attachments/assets/370f9037-cbdc-4ed9-9faa-86585a28fc72">

Anthony

**Comment by 0dan0** [comment]
The older USB trigger demo is only now good on the older cameras, anything that runs the new Blockly code will not run that old style USB wake script. So MAX,H10,11,11mini & 12 need the newer approach.  While new approach does take more code, it is way more powerful and reliable.

What you have will work. You just need a separate ```*WAKE=2``` QR Code to enabled boot on power.

Where is a more complete version of the Dashcam script.
![Basic DashCam V2-7_30_2024, 15_18_16](https://github.com/user-attachments/assets/e459bfa2-ba5d-40b5-b377-7fa50f269bb5)

![image](https://github.com/user-attachments/assets/247463b6-f1fd-4ec2-a32c-bd9fc5f1b55a)
